lib/node_modules/@stdlib/array/base/cusome/lib/assign.js

```diff
@@ -0,0 +1,312 @@
+}
+
+/**
+* Cumulatively tests whether at least n array elements in a provided complex number are truthy and assigns results to provided output array.
+*
+* @private
+* @param {Collection} x - array containing interleaved real and imaginary components
+* @param {integer} n - number of elements to be truthy
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0 ] );
+* var n = 2;
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+* var arr = complex( x, n, arraylike2object( y ), 2, 0 );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns false
+*
+* v = y.get( 6 );
+* // returns false
+*
+* v = y.get( 8 );
+* // returns false
+*/
+function complex( x, n, y, stride, offset ) {
+	var ydata;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	yset = y.accessors[ 1 ];
+	ydata = y.data;
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < x.length; i += 2 ) {
+		if ( flg === false && n > 0 && ( x[ i ] || x[ i+1 ] ) ) {
+			n -= 1;
+		}
+		if (n === 0) {
+			flg = true;
+		}
+		yset( ydata, io, flg );
+		io += stride;
+	}
+	return ydata;
+}
+
+/**
+* Cumulatively tests whether at least n array elements in a provided array are truthy.
+*
+* @private
+* @param {Collection} x - input array
+* @param {integer} n - number of elements to be truthy
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var Uint8Array = require( '@stdlib/array/uint8' );
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Uint8Array( [ 0, 0, 0, 1, 0 ] );
+* var n = 2;
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+*
+* var arr = boolean( x, n, arraylike2object( y ), 2, 0 );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns false
+*
+* v = y.get( 6 );
+* // returns false
+*
+* v = y.get( 8 );
+* // returns false
+*/
+function boolean( x, n, y, stride, offset ) {
+	var ydata;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	yset = y.accessors[ 1 ];
+	ydata = y.data;
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < x.length; i++ ) {
+		if ( flg === false && n > 0 && x[ i ] ) {
+			n -= 1;
+		}
+		if (n === 0) {
+			flg = true;
+		}
```

Comment:
```suggestion
		if ( flg === false && x[ i ] ) {
			n -= 1;
			if ( n === 0 ) {
				flg = true;
			}
		}
```

Here and elsewhere. This avoids needing to check the `if ( n === 0 ) {...}` branch for every iteration, even when `flg` becomes `true`.